### PR TITLE
[18mag] module 1 - supporters

### PR DESCRIPTION
**Pull request by bentziaxl**
Implement module 1 of the new 18mag expansion.
6 supporters (implemented as privates), that are drafted like the minors and majors
adding different abilities that can be used by minors/majors once a round.

### CODE REVIEW

File: lib/engine/game/g_18_mag/step/route.rb [hunk 1]
```diff
@@ -35,8 +35,16 @@ def process_run_routes(action)
...
             super
+            deconvert_plus_one_train if @game.supporters? && @round.original_train
+          end
+
+          def deconvert_plus_one_train
+            @round.ma_plus_train.name = @round.original_train.name
+            @round.ma_plus_train.distance = @round.original_train.distance
+
+            @round.original_train = nil
+            @round.ma_plus_train = nil
```
Comment (roseundy) [inline lib/engine/game/g_18_mag/step/route.rb:47]: What is this doing? Why are you assigning to @round.ma_plus_train and then making it nil?

Comment (bentziaxl) [reply lib/engine/game/g_18_mag/step/route.rb:47]: see here: https://github.com/tobymao/18xx/blob/master/lib/engine/game/g_1822/step/route.rb#L50

I'm modifying the distance of the train that was converted into +1 town range back to it's original values.
That's why i'm change .name and . distance, then setting it to nil.
Similar code to pullman attachment and removal in 1822

**Comment by tobymao** [comment]
@roseundy is this good to go?

**Comment by roseundy** [comment]
Good to go.

**Pull request merged**